### ISSUE TITLE
fileInput <label> element not directly linked to <input>

### ISSUE THREAD

**Issue by acahill** [rstudio/shiny#3426]
Is this a bug in input-file.R? https://github.com/rstudio/shiny/blob/bcb7cde44bc3655b35257fd3cde5c2ee56028148/R/input-file.R#L116

There was a [question on stackoverflow](https://stackoverflow.com/questions/67939403/wcag-2-1-compatibility-with-shinydashboard-app-form-label) where someone was trying to make their fileInput control meet WCAG accessibility requirements. On reading the code in input-file.R I noticed that the tags$label element doesn't seem to be linked to the tags$input element.

The easiest way to link them would be to put the input _inside_ the label (move that closing parenthesis down three lines), or failing that use a [for attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#attr-for)

**Comment by BajczA475** [comment]
I should have mentioned that I think just adding an aria-label to this input element would suffice to silence compliance checkers like WAVE.  In my context, this drop-in JS fix from ChatGPT resolves the issue WAVE flags:

```
suppressFileInputFauxTextbox = function(inputId) {
  tags$script(HTML(sprintf("
    $(document).ready(function() {
      var fileInput = document.getElementById('%s');
      if (!fileInput) return;

      var inputGroup = fileInput.closest('.input-group');
      if (!inputGroup) return;

      var fauxInput = inputGroup.querySelector('input.form-control[type=\"text\"]');
      if (fauxInput) {
        fauxInput.setAttribute('aria-label', 'Filename display only (decorative)');
      }
    });
  ", inputId, inputId, inputId)))
}
```

I include that just to provide more context about what the specific problem is.
